### PR TITLE
Port plugin to Babel 6 plugin API

### LINKED ISSUE #5: Babel 6
As mentioned in https://github.com/ember-cli/ember-cli/issues/5015 this plugin should be ported to the Babel 6 plugin API.

I would have started to port it but then I saw that https://github.com/ember-cli/babel-plugin-feature-flags/pull/4 has rewritten a lot of the code. Should I port the old codebase or wait until https://github.com/ember-cli/babel-plugin-feature-flags/pull/4 is merged?

/cc @mmun @stefanpenner

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves #5 

/cc @mmun @rwjblue @stefanpenner

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,57 +1,63 @@
...
-      } else if (value === 'dynamic') {
-        options.features[feature] = null;
-      } else {
-        throw new Error("An unknown feature state '" + value + "' was detected for '" + feature + "'. Valid values are 'enabled', 'disabled' and 'dynamic'");
-      }
-    }
-  });
+  return {
+    visitor: {
+      Program: function(path, state) {
```
Comment (Turbo87) [inline index.js]: I had to move the options validation into the visitor because the options don't seem to be available before that

**Comment by stefanpenner** [comment]
@Turbo87 looks awesome

**Comment by mmun** [comment]
@Turbo87 My only concern is with closing over `options` in the visitor if there are multiple instances of the plugin running with different options. We can deal with that later though.

**Comment by mmun** [comment]
Done. Let's bump to 0.3.0 before the next release.

**Pull request merged**